Rpi 4.9.y adxl372 bw samp freq support

stefpopa opened the pull request:
These patches add support for the user to select the sampling frequency and the filter bandwidth.

dbogdan commented on drivers/iio/accel/adxl372.c:
```diff
@@ -403,6 +405,18 @@ static int adxl372_get_odr_index(struct adxl372_state *st, int val)
 }
 
+static int adxl372_get_bw_index(struct adxl372_state *st, int val)
+{
```
Even if in this case it's not a real issue, I add this as a general comment: since "int" is platform dependent, I would use a fixed-size type for the frequency variable.

stefpopa replied:
val is passed as an argument to adxl372_get_bw_index() from adxl372_write_raw() which has val parameter defined as type int

dbogdan replied:
Right, in this case you don't have other option. :)

lclausen-adi replied:
For anything that is passed in a register using int is usually OK. The parameter will get passed in a  register no matter what. But if you use a fixed size type, something like int8_t, the compiler will have to add extra code to convert from 8-bit to 32-bit.

commodo commented on drivers/iio/accel/adxl372.c:
```diff
@@ -252,6 +252,8 @@ enum adxl372_fifo_mode {
 	ADXL372_FIFO_OLD_SAVED
 };
 
+static const int adxl372_samp_freq_tbl[5] = {400, 800, 1600, 3200, 6400};
```
nitpick: you could couple the `enum adxl372_odr ` with this information;
something like:
```
static const int adxl372_samp_freq_tbl[5] = {
       [ADXL372_ODR_400HZ] = 400,
       [ADXL372_ODR_800HZ] = 800, 
       [ADXL372_ODR_1600HZ] = 1600, 
       [ADXL372_ODR_3200HZ] = 3200, 
       [ADXL372_ODR_6400HZ] = 6400, 
};
```

commodo replied:
similar comment for `enum adxl372_bandwidth` and `static const int adxl372_bw_freq_tbl`

stefpopa replied:
ack; I will change as you suggested

stefpopa commented:
changelog v1 -> v2:
* coupled `enum adxl372_odr` with `adxl372_samp_freq_tbl`
* coupled `enum adxl372_bandwidth ` with `adxl372_bw_freq_tbl`

commodo commented on drivers/iio/accel/adxl372.c:
```diff
@@ -760,17 +786,49 @@ static int adxl372_write_raw(struct iio_dev *indio_dev,
+		len += scnprintf(buf + len, PAGE_SIZE - len,
+				 "%d ", adxl372_bw_freq_tbl[i]);
+
+	buf[len - 1] = '\n';
```
maybe add a `if (len > 0)` check here [for paranoia reasons];
otherwise this could be an access violation ;

stefpopa replied:
I don't think len = 0 is possible, because even if st->odr is 0, we will still have one character written into buf

commodo replied:
ack;
thanks for the note/clarification

commodo commented:
rest looks fine from my side

stefpopa commented:
changelog v2 ->v3:
* added the .travis.yml script

stefpopa commented:
changelog v3 ->v4:
* used the git commit description style: 'commit <12+ chars of sha1> ("<title line>")'  as a result of checkpatch.pl error

stefpopa commented:
changelog v4 -> v5:
* removed xilinx and pluto targets from the .travis.yml script
* modified the .travis.yml script according to  commit [8df6048cc588 ](https://github.com/analogdevicesinc/linux/pull/64/commits/8df6048cc588bdc7a2fe804549ca635fda2bfefe)(".travis.yml: use TRAVIS_BRANCH env var during pull request") 
* added a new commit which enables adxl372 in adi_bcm2709_defconfig
